tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1311,6 +1359,118 @@ class DxilConf_SM69_Vectorized {
   HLK_TEST(Initialize, float, Vector);
   HLK_TEST(Initialize, double, Vector);
 
+  // Explicit Cast
+
+  HLK_TEST(CastToBool, HLSLBool_t, Vector);
+  HLK_TEST(CastToInt16, HLSLBool_t, Vector);
+  HLK_TEST(CastToInt32, HLSLBool_t, Vector);
+  HLK_TEST(CastToInt64, HLSLBool_t, Vector);
+  HLK_TEST(CastToUint16, HLSLBool_t, Vector);
+  HLK_TEST(CastToUint32, HLSLBool_t, Vector);
+  HLK_TEST(CastToUint64, HLSLBool_t, Vector);
+  HLK_TEST(CastToFloat16, HLSLBool_t, Vector);
+  HLK_TEST(CastToFloat32, HLSLBool_t, Vector);
+  HLK_TEST(CastToFloat64, HLSLBool_t, Vector);
+
+  HLK_TEST(CastToBool, double, Vector);
+  HLK_TEST(CastToInt16, double, Vector);
+  HLK_TEST(CastToInt32, double, Vector);
+  HLK_TEST(CastToInt64, double, Vector);
+  HLK_TEST(CastToUint16_FromFP, double, Vector);
+  HLK_TEST(CastToUint32_FromFP, double, Vector);
+  HLK_TEST(CastToUint64_FromFP, double, Vector);
+  HLK_TEST(CastToFloat16, double, Vector);
+  HLK_TEST(CastToFloat32, double, Vector);
+  HLK_TEST(CastToFloat64, double, Vector);
+
+  HLK_TEST(CastToBool, float, Vector);
+  HLK_TEST(CastToInt16, float, Vector);
+  HLK_TEST(CastToInt32, float, Vector);
+  HLK_TEST(CastToInt64, float, Vector);
+  HLK_TEST(CastToUint16_FromFP, float, Vector);
+  HLK_TEST(CastToUint32_FromFP, float, Vector);
+  HLK_TEST(CastToUint64_FromFP, float, Vector);
+  HLK_TEST(CastToFloat16, float, Vector);
+  HLK_TEST(CastToFloat32, float, Vector);
+  HLK_TEST(CastToFloat64, float, Vector);
+
+  HLK_TEST(CastToBool, HLSLHalf_t, Vector);
+  HLK_TEST(CastToInt16, HLSLHalf_t, Vector);
+  HLK_TEST(CastToInt32, HLSLHalf_t, Vector);
+  HLK_TEST(CastToInt64, HLSLHalf_t, Vector);
+  HLK_TEST(CastToUint16_FromFP, HLSLHalf_t, Vector);
+  HLK_TEST(CastToUint32_FromFP, HLSLHalf_t, Vector);
+  HLK_TEST(CastToUint64_FromFP, HLSLHalf_t, Vector);
+  HLK_TEST(CastToFloat16, HLSLHalf_t, Vector);
+  HLK_TEST(CastToFloat32, HLSLHalf_t, Vector);
+  HLK_TEST(CastToFloat64, HLSLHalf_t, Vector);
+
+  HLK_TEST(CastToBool, uint16_t, Vector);
+  HLK_TEST(CastToInt16, uint16_t, Vector);
+  HLK_TEST(CastToInt32, uint16_t, Vector);
+  HLK_TEST(CastToInt64, uint16_t, Vector);
+  HLK_TEST(CastToUint16, uint16_t, Vector);
+  HLK_TEST(CastToUint32, uint16_t, Vector);
+  HLK_TEST(CastToUint64, uint16_t, Vector);
+  HLK_TEST(CastToFloat16, uint16_t, Vector);
+  HLK_TEST(CastToFloat32, uint16_t, Vector);
+  HLK_TEST(CastToFloat64, uint16_t, Vector);
+
+  HLK_TEST(CastToBool, uint32_t, Vector);
+  HLK_TEST(CastToInt16, uint32_t, Vector);
+  HLK_TEST(CastToInt32, uint32_t, Vector);
+  HLK_TEST(CastToInt64, uint32_t, Vector);
+  HLK_TEST(CastToUint16, uint32_t, Vector);
+  HLK_TEST(CastToUint32, uint32_t, Vector);
```

Comment:
Do these "cast from T to T" ones actually generate any interesting DXIL to validate?

Reply:
Ah! Nope, optimized out. I'll remove those cases.